### ISSUE TITLE
celery chain/group/chord error on 4.0.0 version

### ISSUE THREAD

**Issue by andhdo** [celery/celery#3661]
I have a fragment of code that runs in celery==3.1.24. However in celery==4.0.0 it does not run. See the fragment below:

```
    logger.debug('>> ##LENGTH: %d ' % (len(proc_list)))
    job_workflow = chord([
                scrap_pipeline.s({ 'pk': proc.id, 'task_type': "batch"  },) for proc in proc_list
            ],
            xx_on_finish.s(batchproc)
        )
    

    result = job_workflow.apply_async()

```
## Expected behavior
When i run the code in celery==3.1.24, the header of the chord is executed in parallel similar to a group does, and then es executed the last subtask to notify the finish step of the task.
However, it does not happen in the celery==4.0.0 version.
A similar output is returned using a group inside a chain 

## Actual behavior

[2016-12-07 21:09:12,384: DEBUG/PoolWorker-2] >> ##LENGTH: 2 
[2016-12-07 21:09:12,401: DEBUG/PoolWorker-2] Start from server, version: 0.9, properties: {'capabilities': {'authentication_failure_close': True, 'consumer_cancel_notify': True, 'connection.blocked': True, 'basic.nack': True, 'exchange_exchange_bindings': True, 'publisher_confirms': True, 'consumer_priorities': True, 'per_consumer_qos': True}, 'information': 'Licensed under the MPL.  See http://www.rabbitmq.com/', 'copyright': 'Copyright (C) 2007-2015 Pivotal Software, Inc.', 'version': '3.6.0', 'product': 'RabbitMQ', 'platform': 'Erlang/OTP', 'cluster_name': 'rabbit@rabbitmq'}, mechanisms: ['PLAIN', 'AMQPLAIN'], locales: ['en_US']
[2016-12-07 21:09:12,402: DEBUG/PoolWorker-2] using channel_id: 1
[2016-12-07 21:09:12,403: DEBUG/PoolWorker-2] Channel open
[2016-12-07 21:09:12,409: ERROR/PoolWorker-2] Task m_work.tasks.xx_pipeline[ed169980-52b6-4c1f-836f-d53e6f1194fc] raised unexpected: IndexError('list index out of range',)
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/celery/app/trace.py", line 368, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/celery/app/trace.py", line 623, in __protected_call__
    return self.run(*args, **kwargs)
  File "/app/irj/irj_app/m_work/tasks.py", line 809, in xx_pipeline
    result = job_workflow.apply_async()
  File "/usr/local/lib/python3.5/site-packages/celery/canvas.py", line 1236, in apply_async
    app = self._get_app(body)
  File "/usr/local/lib/python3.5/site-packages/celery/canvas.py", line 1341, in _get_app
    app = tasks[0]._app
IndexError: list index out of range

**Comment by rneilson** [comment]
Also encountering this issue, and I've been banging my head against the wall thinking it was a syntax error on my part...

**Comment by AlexHill** [comment]
Fixed in #3847. Please re-open if you find you still have an issue on the latest master.
